app/controllers/home_page_controller.rb

```diff
@@ -1,4 +1,5 @@
 class HomePageController < ApplicationController
   def index
+    redirect_to controller: 'welcome', action: 'index' unless user_signed_in?
```

Comment:
I suggest you write some generic logic for authorization in the `ApplicationController` and the go with it using `before_action` or in in some other way.

Reply:
I think we can return a 401 error if a user is unauthenticated.